tls/s2n_resume.c

```diff
@@ -155,3 +224,58 @@ int s2n_store_to_cache(struct s2n_connection *conn)
 
     return 0;
 }
+
+int s2n_connection_set_session(struct s2n_connection *conn, const uint8_t *session, size_t length)
+{
+
+    notnull_check(conn);
+    notnull_check(session);
+
+    struct s2n_blob session_data;
+    GUARD(s2n_alloc(&session_data, length));
+    memcpy_check(session_data.data, session, length);
```

Comment:
hm, why do you need to copy the buffer? Can't we deserialize without copying?

Reply:
Yeah the problem is if I assign the session to blob directly I get:
cast discards qualifiers from pointer target type warning and I did not want to use non const pointer for session in the api.

Reply:
ah, that's annoying... I guess it's fine for now, but I'd like some solution for this problem in future to avoid copying data over when it's not really necessary. (I.e. allowing const cast in certain scenarios, not using consts at all, introducing const blobs...)